### PR TITLE
Add Honda Insight

### PR DESCRIPTION
**Pull request by energee**
We should be able to test this next week

### CODE REVIEW

File: selfdrive/car/honda/values.py [hunk 1]
```diff
@@ -108,6 +112,7 @@ class CAR:
   CAR.CIVIC_HATCH: dbc_dict('honda_civic_hatchback_ex_2017_can_generated', None),
   CAR.CRV: dbc_dict('honda_crv_touring_2016_can_generated', 'acura_ilx_2016_nidec'),
   CAR.CRV_5G: dbc_dict('honda_crv_ex_2017_can_generated', None),
+  CAR.INSIGHT: dbc_dict('honda_insight_ex_2019_can_generated.dbc'),
```
Comment (vanillagorillaa) [inline selfdrive/car/honda/values.py]:
```suggestion
  CAR.INSIGHT: dbc_dict('honda_insight_ex_2019_can_generated.dbc', None),
```

**Comment by rbiasini** [comment]
cool!

**Comment by energee** [comment]
@rbiasini Let me know if 7be6fa5 works for you

**Comment by rbiasini** [comment]
@energee https://github.com/commaai/openpilot/commit/7be6fa54a5e0fa21bc0e05e708cba2b5f3250e17 looks good. Is this PR ready to go? I will wait to merge after 0.5.6 gets released today or tmrw.

**Comment by energee** [comment]
He thought the steer max warning was coming on often on turns but i guess this might be expected, depending upon his definition of a turn. Can you tell if there is any obvious tuning needed from the logs?

**Comment by energee** [comment]
Rebased

**Comment by rbiasini** [comment]
Reviewed @energee ... the alert is sent correctly, hard to say if it is less capable than other Hondas. Have you guys tried commanding more steering torque?

Still, I can merge as is.

**Comment by rbiasini** [comment]
Oh, the car addition to the README is needed :)

**Comment by rbiasini** [comment]
Actually, there is a more serious problem. This fingerprint fully contains the Accord Hybrid fingerprint, so the Accord Hybrid will not be fingerprintable.

**Comment by energee** [comment]
oh, nice catch, any ideas on how to deal with this?

**Comment by rbiasini** [comment]
@energee discussing this internally. Maybe worth to just consider it the same as the ACCORD HYBRID

**Comment by dustinfields** [comment]
@rbiasini The Accord and Insight have different parameters according to @energee

**Comment by rbiasini** [comment]
@dustinfields @energee: yeah... I don't see a good solution unfortunately.

**Comment by energee** [comment]
I have an idea for this, will probably get to it this weekend

**Comment by theantihero** [comment]
Is there any update to this yet? Also, it seems like OP and Insight seems to engage on 11 on giraffe instead of 10. It seems to have full stop and go without pressing resume on the wheel. Seems weird, is that normal?

**Comment by energee** [comment]
@theantihero That is weird, could be an issue with your giraffe but if that were the case, 11 or 10 would likely function the same.

**Comment by theantihero** [comment]
> @theantihero That is weird, could be an issue with your giraffe but if that were the case, 11 or 10 would likely function the same.

Hmm possibly like you said. 10 makes my car error out on the dash for LKAS, ACC, Brake System, TPMS, etc. I just tried driving with 11 during daytime since my last recording was during night.

For the nighttime one i think after halfway through video is when it's a straight road with OP engaged.

**Night**: https://community.comma.ai/cabana/?route=fb51d190ddfd8a90%7C2019-01-10--03-17-10&max=14&url=https%3A%2F%2Fchffrprivate.blob.core.windows.net%2Fchffrprivate3%2Fv2%2Ffb51d190ddfd8a90%2Fd93535ab202ad33aacf0f5142fed516c_2019-01-10--03-17-10

**Day**: https://community.comma.ai/cabana/?route=fb51d190ddfd8a90%7C2019-01-10--16-15-52&max=19&url=https%3A%2F%2Fchffrprivate.blob.core.windows.net%2Fchffrprivate3%2Fv2%2Ffb51d190ddfd8a90%2F4b1933316f6c8f422bbd14e9728781f6_2019-01-10--16-15-52

Question. Is the steering limit error popping up frequently normal as well? Steering limit seems rly low even on small curving roads on a straight road.

**Comment by energee** [comment]
Stop in Philly and maybe I finish port :)

**Comment by theantihero** [comment]
Ill try to ;)

**Comment by groovin20** [comment]
I apologize. I know this is probably not the correct forum for this question but can you point me to the correct hardware to get for the Insight? Which giraffe connector is the correct one for the Insight?

**Comment by energee** [comment]
@groovin20 EON https://comma.ai/shop/products/eon-gold-dashcam-devkit?ref=3
Panda (Gray or white) https://comma.ai/shop/products/panda-obd-ii-dongle?ref=3
Giraffe (Bosch) https://comma.ai/shop/products/giraffe?ref=3

**Comment by vanillagorillaa** [comment]
There is also this guide for future reference, https://comma.ai/giraffe-selection

**Comment by dustinfields** [comment]
Also when you join discord hit me up we have a better port for the insight.

Ironworm#8498

**Comment by groovin20** [comment]
Thank you so much for the attention!! I checked the links but I am still confused about which Giraffe to order. Do any of you know which of the 3 Honda options work for the Insight? The link that says there are pictures at the bottom is not showing any pictures.

**Comment by dustinfields** [comment]
It's the Bosch one you want

**Comment by groovin20** [comment]
dustinfields, what is discord? I received everything in the mail. I hooked everything up. I am downloading open pilot now. I am really excited about this!

**Comment by mtourangeau** [comment]
Discord is a messaging application.

**Comment by groovin20** [comment]
Ok. I will join now.

**Comment by groovin20** [comment]
I have downloaded the new openpilot. I am not sure how to engage steering assist lower than 45mph. Is there something else I am supposed to add?

**Comment by dustinfields** [comment]
Did you join discord? What is your username?

**Comment by groovin20** [comment]
yes groovin20#1955

**Comment by groovin20** [comment]
I sent a friend request.

**Comment by Kiril-Lange** [comment]
Any update on this? I just got an Insight and will be looking to set up OP in the coming months.

**Comment by dustinfields** [comment]
This PR is out of date, we have a newest OP running.

Join discord and send me a friends request dustinfields

**Comment by energee** [comment]
Rebased against latest code, in preparation for black panda support.

@dustinfields Do you have an insight?

**Comment by dustinfields** [comment]
@energee Yes I do I am Ironworm on Discord.

**Comment by energee** [comment]
Try this branch, it should work as-is with no changes, and i will keep it rebased until it is ready to be merged.

Once we can get it upstreamed with black panda support, a legacy hardware fork will be really simple, probably single line change.

**Comment by chrisj951** [comment]
This branch no longer works for me. @dustinfields  does it work for you? I just get LKAS / ACC errors.

**Comment by dustinfields** [comment]
@chrisj951 this is an old branch add me on discord we have a custom fork for the insight

**Comment by rafaellichen** [comment]
@energee I have a Insight 2019; really want to try the OP out with it. Is this PR going to be updated? Is the fingerprint issue resolved yet? Thanks ;)

**Comment by theantihero** [comment]
Im still able to enable OP by pressing my paddles lol. Also just hit us up
discord we have a insight group. Works on 0.7.0.

**Comment by dustinfields** [comment]
@rafaellichen  Add me on discord 

Ironworm#8498

**Comment by coleyb** [comment]
I'd love to be added to the Discord group as well! Coleybrown#3439

**Pull request closed without merging**